network-libp2p/src/discovery/handler.rs

```diff
@@ -374,6 +376,10 @@ impl ConnectionHandler for Handler {
                     }
 
                     self.state = HandlerState::ReceiveHandshake;
+                    self.state_timeout
+                        .as_mut()
+                        .expect("state timeout must exist")
+                        .reset(Duration::from_millis(Self::STATE_TRANSITION_TIMEOUT_MS));
```

Comment:
Just another option without an `expect`. If you want to make sure there was a timeout for the previous state, too, then your version is better.
```suggestion
                    self.state_timeout = Some(Delay::new(Duration::from_millis(
                        Self::STATE_TRANSITION_TIMEOUT_MS,
                    )));
```